xclim/indices/run_length.py

```diff
@@ -1683,3 +1690,144 @@ def suspicious_run(
+            starts[starts < time_min] = np.nan
+            return starts
+
+        ds["event_start"] = xr.apply_ufunc(
+            _get_start_cftime,
+            ds.event_start.fillna(-1),
+            dask="parallelized",
+            kwargs={"time_min": time_min.item()},
+            output_dtypes=[time_min.dtype],
+        )
+    else:
+        ds["event_start"] = ds.event_start.copy(
+            data=time_min.values + ds.event_start.data.astype("timedelta64[s]")
+        )
+
+    ds["event"] = np.arange(1, ds.event.size + 1)
+    ds["event_length"].attrs["units"] = "1"
+    ds["event_effective_length"].attrs["units"] = "1"
+    ds["event_start"].attrs["units"] = ""
+    if data is not None:
+        ds["event_sum"].attrs["units"] = data.units
+    return ds
+
+
+# TODO: Implement more event stats ? (max, effective sum, etc)
+def find_events(
+    condition: xr.DataArray,
+    window: int,
+    condition_stop: xr.DataArray | None = None,
+    window_stop: int | None = None,
+    data: xr.DataArray | None = None,
+    freq: str | None = None,
+):
+    """Find events (runs).
+
+    An event starts with a run of ``window`` consecutive True values in the condition
+    and stops with ``window_stop`` consecutive True values in the stop condition.
+
+    This returns a Dataset with each event along an `event` dimension. It does not
+    perform statistics over the events like other function in this module do.
+
+    Parameters
+    ----------
+    condition : DataArray of boolean values
+        The boolean mask, true where the start condition of the event is fulfilled.
+    window: int
+        The number of consecutive True values for an event to start.
+    condition_stop : DataArray of boolean values, optional
+        The stopping boolean mask, true where the end condition of the event is fulfilled.
+        Defaults to the opposite of ``condition``.
+    window_stop : int, optional
+        The number of consecutive True values in ``condition_stop`` for an event to end.
+        Defaults to ``window``.
+    data: DataArray, optional
+        The actual data. If present, its sum within each event is added to the output.
+    freq: str, optional
+        A frequency to divide the data into periods. If absent, the output has not time dimension.
+        If given, the events are searched within in each resample period independently.
+
+    Returns
+    -------
+    xr.Dataset, same shape as the data it has a new "event" dimension (and the time dimension is resample or removed, according to ``freq``).
+        event_length: The number of time steps in each event
+        event_effective_length: The number of time steps of even event where the start condition is true.
+        event_start: The datetime of the start of the run.
+        event_sum: The sum within each event, only considering the steps where start condition is true. Only present if ``data`` is given.
+    """
+    window_stop = window_stop or window
```

Comment:
Do we want the default behaviour to reproduce a regular run length, or reproduce the concept of season? 

If the concept of season, then it's all right. If run_length, then we should take `window_stop=1`. I think both are fine, but we should add a note in the docstring to explain what the default behaviour reproduces

Reply:
Ah! You're right I didn't think of that. I was thinking of run-like, so I'll change to 1. And edit the doc.